src/parameter.jl

```diff
@@ -60,6 +60,7 @@ struct Parameter{T, F, InPlace}
     Parameter(f::F, val::T, model) where {T, F} = addparameter!(model, new{T, F, true}(Ref(true), f, Base.RefValue(val)))
 end
 Parameter(f, model) = Parameter{typeof(f())}(f, model)
+Parameter(A::AbstractArray, model) = Parameter(identity, A, model)
```

Comment:
I don't think this is a good idea. I understand there is a need for an easier way (easier than `Parameter(identity, val, model)` to defer evaluation (#67), but special-casing `AbstractArray` like this I think is kind of confusing for users.

Ideally, I'd prefer this to be done in a `Parameter` constructor, but since there aren't really any constructor signatures available between the in-place and out-of-place versions, maybe we do need a separate `defer` function then. Or maybe we should get rid of the constructor on the line above this one as it's prone to type inference issues anyway (although that's much less of an issue now with the wrapping), and use that for this purpose. Any more thoughts, @rdeits?

Reply:
Yeah, I think it's not a good idea to have a special `Parameter` constructor for particular types. I still like `defer(x, model)`, as it's pretty explicit about what it's doing (more so than `Parameter(identity, x, model)`, I think).